flutter/windows/runner/win32_window.cpp

```
@@ -15,6 +16,43 @@ constexpr const wchar_t kWindowClassName[] = L"FLUTTER_RUNNER_WIN32_WINDOW";
+  wchar_t exe_path[MAX_PATH];
+  if (!GetModuleFileNameW(nullptr, exe_path, MAX_PATH)) {
+    return nullptr;
+  }
+
+  std::wstring icon_path = exe_path;
+  size_t last_slash = icon_path.find_last_of(L"\\/");
+  if (last_slash == std::wstring::npos) {
+    return nullptr;
+  }
+
+  icon_path = icon_path.substr(0, last_slash + 1);
```

Cppcheck (2.19.0)

[performance] Ineffective call of function 'substr' because a prefix of the string is assigned to itself. Use resize() or pop_back() instead.

(uselessCallsSubstr)